python/ray/data/datasource/datasink.py

```diff
@@ -50,7 +49,32 @@ def on_write_complete(self, write_results: List[Any]) -> None:
             write_results: The objects returned by every
                 :meth:`~ray.data.Datasink.write` task.
         """
-        pass
+        total_num_rows = 0
+        total_size_bytes = 0
+
+        for result in raw_write_results:
+            # `result` is a Block containing a single row with write results/stats.
+            # Calling `iter_rows()` is the cleanest way to extract a single row
+            # for various block types.
+            for row in BlockAccessor.for_block(result).iter_rows(None):
+                write_results = row
+
+            total_num_rows += write_results["write_num_rows"]
+            total_size_bytes += write_results["write_size_bytes"]
+
+        aggregated_results = {
+            "total_num_rows": total_num_rows,
+            "total_size_bytes": total_size_bytes,
+        }
+
+        aggregated_results_str = ""
+        for k, v in aggregated_results.items():
+            aggregated_results_str += f"\t{k}: {v}\n"
+
+        logger.info(
+            f"`write_datasink()` succeeded. Aggregated write results:\n"
```

Comment:
```suggestion
            f"Write operation succeeded. Aggregated write results:\n"
```
`write_datasink` might be confusing if users are using a high-level API, e.g. `write_parquet`.